cypress/component/basic/typescript/ts-spec.tsx

```diff
@@ -1,13 +1,34 @@
 import React from 'react'
 import { mount } from 'cypress-react-unit-test'
-
-const Button = ({ children, ...rest }) => {
-  return <button {...rest}>{children}</button>
-}
+import { Button } from './Button'
+// import the same Button component using path alias
+// that should be set up in webpack AND in tsconfig
+// https://glebbahmutov.com/blog/using-ts-aliases-in-cypress-tests/
+import { Button as SameButton } from '@basic/typescript/Button'
 
 describe('Component spec in typescript', () => {
   it('works', () => {
     mount(<Button>Button Label</Button>)
     cy.contains('button', 'Button Label').should('be.visible')
   })
+
+  it('calls passed on click prop', () => {
+    // TODO how to type this?
+    // @ts-ignore
+    mount(<Button onClick={cy.stub().as('click')}>Click Me</Button>)
```

Comment:
not sure how to provide optional `onClick` amongst the `...props` @dmtrKovalenko